### PR TITLE
Add automatic dark theme support

### PR DESCRIPTION
**Pull request by Phyxius**
This PR adds an automatic dark theme. I took the color specific rules out of the original CSS files, inverted the colors, and created new CSS files containing those specific rules. The new files are bundled by Hugo into a single dark stylesheet, which is embedded in the `HEAD` after the main one with a media query to instruct browsers to load it if the user is set to a dark theme. Browsers that don't support that media query should just ignore the sheet, so there shouldn't be any compatibility impact.

### CODE REVIEW

File: layouts/partials/head.html [hunk 1]
```diff
@@ -16,8 +16,14 @@
   {{ $style := resources.Get "css/style.css" }}
   {{ $markdown := resources.Get "css/markdown.css" }}
   {{ $syntaxHighlight := resources.Get "css/syntax-highlight.css" }}
-  {{ $css := slice $style $markdown $syntaxHighlight | resources.Concat "style.css" | resources.Minify | resources.Fingerprint }}
-  <link rel="stylesheet" href="{{ $css.Permalink }}">
+  {{ $syntaxHighlightDark := resources.Get "css/syntax-highlight-dark.css" }}
+  {{ $css := slice $style $markdown $syntaxHighlight $syntaxHighlightDark | resources.Concat "style.css" | resources.Minify | resources.Fingerprint }}
```
Comment (koirand) [inline layouts/partials/head.html]: Isn't `$syntaxHighlightDark` unnecessary? In light mode, the text color will be incorrect.

File: layouts/partials/head.html [hunk 2]
```diff
@@ -16,8 +16,14 @@
...
+  <link rel="stylesheet" href="{{ $css.Permalink }}" integrity="{{ $css.Data.Integrity }}">
+  {{ $styleDark := resources.Get "css/style-dark.css" }}
+  {{ $markdownDark := resources.Get "css/markdown-dark.css" }}
+  {{ $syntaxHighlightDark := resources.Get "css/syntax-highlight-dark.css" }}
```
Comment (koirand) [inline layouts/partials/head.html]: There are two definitions of `$syntaxHighlightDark`. Can they be one?

File: assets/css/syntax-highlight-dark.css [hunk 3]
```diff
@@ -0,0 +1,59 @@
+/* Background */ .chroma { color: #f8f8f2; background-color: #272822 }
```
Comment (koirand) [inline assets/css/syntax-highlight-dark.css]: The `.chroma` class is not used in HTML. Could you use `.highlight` instead?

**Comment by Phyxius** [comment]
The dark syntax highlighting CSS turned out to be completely unnecessary so I removed it

**Comment by Phyxius** [comment]
Also code blocks look weird in either mode (you get this weird white artifacting) if they have rounded corners, so I think removing the rounded corners on them is the way to go.

**Comment by koirand** [comment]
Tank you!

**Pull request merged**